From a14d345c6dcc231354b98228056cf5b442712681 Mon Sep 17 00:00:00 2001
From: Yi Sun <yi.sun@intel.com>
Date: Mon, 28 Oct 2024 18:45:52 +0800
Subject: [PATCH] accel-config: Refine the Usage of enable/disable-device
'Content-type:text/plain'

Add the usage for newly added options of enable-device and disable-device.

Signed-off-by: Yi Sun <yi.sun@intel.com>
---
 .../accfg/accel-config-disable-device.txt        |  5 +++++
 .../accfg/accel-config-enable-device.txt         |  5 +++++
 accfg/enable.c                                   | 16 ++++++++++++++--
 3 files changed, 24 insertions(+), 2 deletions(-)

diff --git a/Documentation/accfg/accel-config-disable-device.txt b/Documentation/accfg/accel-config-disable-device.txt
index 8952980e3468..0662092a785b 100644
--- a/Documentation/accfg/accel-config-disable-device.txt
+++ b/Documentation/accfg/accel-config-disable-device.txt
@@ -11,10 +11,15 @@ SYNOPSIS
 --------
 [verse]
 'accel-config disable-device' <device>
+'accel-config disable-device' <device-type>
+	dsa: disable all DSA devices
+	iax: disable all IAX devices
+	all: disable all devices
 
 EXAMPLE
 -------
 accel-config disable-device dsa0
+accel-config disable-device all
 
 OPTIONS
 -------
diff --git a/Documentation/accfg/accel-config-enable-device.txt b/Documentation/accfg/accel-config-enable-device.txt
index 21aba987652f..8da29065d60e 100644
--- a/Documentation/accfg/accel-config-enable-device.txt
+++ b/Documentation/accfg/accel-config-enable-device.txt
@@ -11,10 +11,15 @@ SYNOPSIS
 --------
 [verse]
 'accel-config enable-device' <device>
+'accel-config enable-device' <device-type>
+	dsa: enable all configured DSA devices
+	iax: enable all configured IAX devices
+	all: enable all configured devices
 
 EXAMPLE
 -------
 accel-config enable-device dsa0
+accel-config enable-device all
 
 include::../copyright.txt[]
 
diff --git a/accfg/enable.c b/accfg/enable.c
index 4e7d716ff95c..18b560a73646 100644
--- a/accfg/enable.c
+++ b/accfg/enable.c
@@ -285,7 +285,13 @@ static int wq_action(int argc, const char **argv, const char *usage,
 int cmd_disable_device(int argc, const char **argv, void *ctx)
 {
 	char *usage =
-	    "accel-config disable-device <accel_basename0> [<accel_basename1>..<accel_basenameN>] [<options>]";
+	    "\naccel-config disable-device <accel_basename0> [<accel_basename1>..<accel_basenameN>] [<options>]\n"
+	    "accel-config disable-device <device type>\n"
+	    "    device_type: can be one of following values\n"
+	    "        dsa: disable all DSA devices\n"
+	    "        iax: disable all IAX devices\n"
+	    "        all: disable all devices\n";
+
 	int count = device_action(argc, argv, usage, device_disable_options,
 				  DEV_ACTION_DISABLE, ctx);
 	return count >= 0 ? 0 : EXIT_FAILURE;
@@ -294,7 +300,13 @@ int cmd_disable_device(int argc, const char **argv, void *ctx)
 int cmd_enable_device(int argc, const char **argv, void *ctx)
 {
 	char *usage =
-	    "accel-config enable-device <accel_basename0> [<accel_basename1>..<accel_basenameN>] [<options>]";
+	    "\naccel-config enable-device <accel_basename0> [<accel_basename1>..<accel_basenameN>] [<options>]\n"
+	    "accel-config enable-device <device type>\n"
+	    "    device_type: can be one of following values\n"
+	    "        dsa: enable all configured DSA devices\n"
+	    "        iax: enable all configured IAX devices\n"
+	    "        all: enable all configured devices\n";
+
 	int count = device_action(argc, argv, usage, device_options,
 				  DEV_ACTION_ENABLE, ctx);
 	return count >= 0 ? 0 : EXIT_FAILURE;
-- 
2.48.0

